python/triton/tools/link.py

```diff
@@ -36,7 +38,7 @@ def __init__(self) -> None:
         # [name, hash, suffix]
         self.kernel_name = re.compile("^([\\w]+)_([\\w]+)_([\\w]+)$")
         # [(type, name)]
-        self.c_sig = re.compile("[\\s]*(\\w+)\\s(\\w+)[,]?")
+        self.c_sig = re.compile(r"\s*(\w+\*?)\s+(\w+)[,]?\s*")
```

Comment:
The regex is different from the original. Can you update a comment (or add an example) for what we are matching. Also:
```suggestion
        self.c_sig = re.compile(r"\s*(\w+\*?)\s+(\w+),?\s*")
```